data-prepper-plugins/otel-proto-common/src/main/java/com/amazon/dataprepper/plugins/otel/codec/OTelProtoCodec.java

```diff
@@ -457,4 +486,294 @@ protected AnyValue objectToAnyValue(final Object obj) throws UnsupportedEncoding
+                if (i == 0) {
+                    double min = -Float.MAX_VALUE; // "-Infinity"
+                    double max = explicitBoundsList.get(i);
+                    Long bucketCount = bucketCountsList.get(i);
+                    buckets.add(new DefaultBucket(min, max, bucketCount));
+                } else if (i == bucketCountsList.size() - 1) {
+                    double min = explicitBoundsList.get(i - 1);
+                    double max = Float.MAX_VALUE; // "Infinity"
+                    Long bucketCount = bucketCountsList.get(i);
+                    buckets.add(new DefaultBucket(min, max, bucketCount));
+                } else {
+                    double min = explicitBoundsList.get(i - 1);
+                    double max = explicitBoundsList.get(i);
+                    Long bucketCount = bucketCountsList.get(i);
+                    buckets.add(new DefaultBucket(min, max, bucketCount));
+                }
+            }
+        }
+        return buckets;
+    }
+
+    /**
+     * Converts a List of {@link io.opentelemetry.proto.metrics.v1.Exemplar} values to {@link DefaultExemplar}, the
+     * internal representation for Data Prepper
+     *
+     * @param exemplarsList the List of Exemplars
+     * @return a mapped list of DefaultExemplars
+     */
+    public static List<Exemplar> convertExemplars(List<io.opentelemetry.proto.metrics.v1.Exemplar> exemplarsList) {
+        return exemplarsList.stream().map(exemplar ->
+                        new DefaultExemplar(convertUnixNanosToISO8601(exemplar.getTimeUnixNano()),
+                                getExemplarValueAsDouble(exemplar),
+                                Hex.encodeHexString(exemplar.getSpanId().toByteArray()),
+                                Hex.encodeHexString(exemplar.getTraceId().toByteArray()),
+                                unpackExemplarValueList(exemplar.getFilteredAttributesList())))
+                .collect(Collectors.toList());
+    }
+
+
+    static double[] calculateBoundaries(int scale) {
+        int len = 1 << Math.abs(scale);
+        double[] boundaries = new double[len + 1];
+        for (int i = 0; i <= len ; i++) {
+            boundaries[i] = scale >=0 ?
+                    Math.pow(2., i / (double) len) :
+                    Math.pow(2., Math.pow(2., i));
+        }
+        return boundaries;
+    }
+
+    /**
+     * Maps a List of {@link io.opentelemetry.proto.metrics.v1.ExponentialHistogramDataPoint.Buckets} to an
+     * internal representation for Data Prepper.
+     * See <a href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#exponential-buckets">data model</a>
+     *
+     * @param buckets the list of buckets
+     * @param scale the scale of the exponential histogram
+     * @return a mapped list of Buckets
+     */
+    public static List<Bucket> createExponentialBuckets(ExponentialHistogramDataPoint.Buckets buckets, int scale) {
```

Comment:
Should this method have any assertions on the `scale` to avoid issues with large values? With scale=30 the data point would have 1 billion buckets.